Source/WebCore/page/Navigation.cpp

```diff
@@ -706,7 +706,10 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
         if (navigationType == NavigationNavigationType::Traverse)
             m_suppressNormalScrollRestorationDuringOngoingNavigation = true;
 
-        // FIXME: Step 7: URL and history update
+        if (navigationType == NavigationNavigationType::Push || navigationType == NavigationNavigationType::Replace) {
+            auto historyHandling = navigationType == NavigationNavigationType::Replace ? NavigationHistoryBehavior::Replace : NavigationHistoryBehavior::Push;
```

Comment:
Awkward that we need to translate `Push` to `Push` and `Replace` to `Replace`. Wish there was a more elegant way to write this.

Reply:
Yes, and also StateObjectType enum by now seems superfluous. So, if Navigation API becomes more accepted, I guess it will make sense to use more of its enums and constants, like I think happened for Fetch API.